lib/logging/log.js

```diff
@@ -0,0 +1,549 @@
+var extend = require('extend');
+var format = require('string-format-obj');
+var is = require('is');
+var JSONStream = require('JSONStream');
+var nodeutil = require('util');
+var propAssign = require('prop-assign');
+var pumpify = require('pumpify');
+var streamEvents = require('stream-events');
+var through = require('through2');
+
+/**
+ * @type {module:logging/entry}
+ * @private
+ */
+var Entry = require('./entry.js');
+
+/**
+ * @type {module:common/serviceObject}
+ * @private
+ */
+var ServiceObject = require('../common/service-object.js');
+
+/**
+ * A log is a named collection of entries, each entry representing a timestamped
+ * event. Logs can be produced by Google Cloud Platform services, by third-party
+ * services, or by your applications. For example, the log `apache-access` is
+ * produced by the Apache Web Server, but the log
+ * `compute.googleapis.com/activity_log` is produced by Google Compute Engine.
+ *
+ * @resource [Introduction to Logs]{@link https://cloud.google.com/logging/docs/api/#logs}
+ *
+ * @alias module:logging/log
+ * @constructor
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var logging = gcloud.logging();
+ * var log = logging.log('syslog');
+ */
+function Log(logging, name) {
+  var methods = {
+    /**
+     * Delete the log.
+     *
+     * @resource [projects.logs.delete API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v2beta1/projects.logs/delete}
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * log.delete(function(err, apiResponse) {
+     *   if (!err) {
+     *     // The log was deleted.
+     *   }
+     * });
+     */
+    delete: true
+  };
+
+  this.name = name.replace(/^.*logs\//, '');
+  this.formattedName_ = Log.formatName_(logging.projectId, name);
+
+  ServiceObject.call(this, {
+    parent: logging,
+    baseUrl: '/logs',
+    id: this.name,
+    methods: methods
+  });
+}
+
+nodeutil.inherits(Log, ServiceObject);
+
+/**
+ * Return an array of log entries with the desired severity assigned.
+ *
+ * @private
+ *
+ * @param {object|object[]} entries - Log entries.
+ * @param {string} severity - The desired severity level.
+ */
+Log.assignSeverityToEntries_ = function(entries, severity) {
+  var entriesClone = [].slice.call(entries);
+  return arrify(entriesClone).map(propAssign('severity', severity));
+};
+
+/**
+ * Format the name of a log. A log's full name is in the format of
+ * 'projects/{projectId}/logs/{logName}'.
+ *
+ * @private
+ *
+ * @return {string}
+ */
+Log.formatName_ = function(projectId, name) {
+  if (name.indexOf('logs/') > -1) {
```

Comment:
Based on the method definition, yes. Nothing stops the user to give invalid data?
(unless the user cannot call this method directly, in which case why not say that name is only the  part after projects/*/logs/ ?